include/pulsar/Producer.h

```diff
@@ -23,13 +23,16 @@
 #include <pulsar/defines.h>
 #include <stdint.h>
 
+#include <functional>
 #include <memory>
 
 namespace pulsar {
 class ProducerImplBase;
 class PulsarWrapper;
 class PulsarFriend;
 
+typedef std::function<void(Result, const MessageId& messageId)> SendCallback;
```

Comment:
You don't need to redefine these here. Remove the import of `Producer.h` from `ProducerInterceptor.h` and the compilation will succeed